### PR TITLE
feat: 🎸 [JIRA: HCPSDKFIORIUIKIT-1904] create a new empty state view.

### PR DESCRIPTION
**Pull request by AllenCao001**

### CODE REVIEW

**Review by billzhou0223 — CHANGES_REQUESTED** [review]
You need to follow the pattern for create new components. 

Docs are here https://github.com/SAP/cloud-sdk-ios-fiori/blob/main/CONTRIBUTING.md#core-component-development.

**Comment by AllenCao001** [comment]
> You need to follow the pattern for create new components.
> 
> Docs are here https://github.com/SAP/cloud-sdk-ios-fiori/blob/main/CONTRIBUTING.md#core-component-development.

From the outset, I implement the `EmptyStateView` refer to `NoDataView` in `FioriCharts` module, and likewise place the new file in this. Whereas I didn't find the use of any pattern under the module, and just found use in the `FiorSwiftUICore` module, so which one module that the `EmptyStateView` could be placed in? And Is it necessary to use the complex pattern 
for a simple view?

**Comment by billzhou0223** [comment]
> > You need to follow the pattern for create new components.
> > Docs are here https://github.com/SAP/cloud-sdk-ios-fiori/blob/main/CONTRIBUTING.md#core-component-development.
> 
> From the outset, I implement the `EmptyStateView` refer to `NoDataView` in `FioriCharts` module, and likewise place the new file in this. Whereas I didn't find the use of any pattern under the module, and just found use in the `FiorSwiftUICore` module, so which one module that the `EmptyStateView` could be placed in? Is it necessary to use the complex pattern for a simple view?

Reusable components in general should be placed in FioriSwiftUICore module.

**Comment by AllenCao001** [comment]
@billzhou0223 I restart to create an empty state view by the sourcery pattern, please review it again.

File: Sources/FioriSwiftUICore/Models/ModelDefinitions.swift [hunk 1]
```diff
@@ -330,3 +330,13 @@ public protocol SignatureCaptureViewModel: AnyObject {
     // sourcery: no_view
     var onDelete: (() -> Void)? { get }
 }
+
+// sourcery: generated_component_composite
+public protocol EmptyStateViewModel: TitleComponent, DescriptionTextComponent {
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/Models/ModelDefinitions.swift:335]: It should conform to `DetailImageComponent` instead of adding image as an property.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]
Please fix all CI failures.

File: Sources/FioriSwiftUICore/Models/ModelDefinitions.swift (hunk 1 above)
Comment (AllenCao001) [reply Sources/FioriSwiftUICore/Models/ModelDefinitions.swift:335]: At first, it conforms to `IconComponent`, but I can't append a `.resizable()` modifier to it, like the below:
```
struct Icon: ViewModifier {
    func body(content: Content) -> some View {
         content
//                  .resizable()
                    .aspectRatio(contentMode: .fit)
                    .frame(width: 160, height: 160)
            }
        }
```
so I changed it to use an Image.

Comment (billzhou0223) [reply Sources/FioriSwiftUICore/Models/ModelDefinitions.swift:335]: You don't need to apply resizable modifier in SDK. Developers will apply it if they want it to be resizable. If we do it in SDK, there is no way to disable it from app side.

Comment (AllenCao001) [reply Sources/FioriSwiftUICore/Models/ModelDefinitions.swift:335]: ok.

File: Sources/FioriSwiftUICore/Views/EmptyStateView+View.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+
+extension Fiori {
+    enum EmptyStateView {
+        struct Title: ViewModifier {
```
[the remark this answers is not in the document]
Comment (AllenCao001) [reply Sources/FioriSwiftUICore/Views/EmptyStateView+View.swift:5]: > Nesting Violation: Types should be nested at most 1 level deep (nesting)

@billzhou0223 I'm confused about why the error occurred. The above code is created automatically and refers to other files, such as `WelcomeScreen+View.swift` and so on.

**Comment by billzhou0223** [comment]
You can ignore this for now

Best,
Bill


From: C5342925 ***@***.***>
Date: Tuesday, November 29, 2022 at 5:46 PM
To: SAP/cloud-sdk-ios-fiori ***@***.***>
Cc: Zhou, Bill ***@***.***>, Mention ***@***.***>
Subject: Re: [SAP/cloud-sdk-ios-fiori] feat: 🎸 [JIRA: HCPSDKFIORIUIKIT-1904] create a new empty state view. (PR #489)

@AllenCao001 commented on this pull request.

________________________________

In Sources/FioriSwiftUICore/Views/EmptyStateView+View.swift<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSAP%2Fcloud-sdk-ios-fiori%2Fpull%2F489%23discussion_r1035449438&data=05%7C01%7Cbill.zhou01%40sap.com%7C91c5e465014b4f6ec8f308dad274b815%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638053696010863232%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=wUzM1G6Y%2FV9HpVNbhFIeLiNFXwglT6svY5TWsvaq1IM%3D&reserved=0>:

> @@ -0,0 +1,65 @@

+import SwiftUI

+

+extension Fiori {

+    enum EmptyStateView {

+        struct Title: ViewModifier {

Nesting Violation: Types should be nested at most 1 level deep (nesting)

@billzhou0223<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbillzhou0223&data=05%7C01%7Cbill.zhou01%40sap.com%7C91c5e465014b4f6ec8f308dad274b815%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638053696010863232%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Bpe%2Bv1EcuHGIAWO8YmkvW3Iukt10v9gfjAezM%2Fza2Sg%3D&reserved=0> I'm confused about why the error occurred. The above code is created automatically and refers to other files, such as WelcomeScreen+View.swift and so on.

**Review by billzhou0223 — APPROVED** [review]

**Pull request merged**